6.x/crud-fields.md

```diff
@@ -943,63 +943,37 @@ Input preview:
 
-**How it works:**
+- **Option 2.** Handle the upload process yourself. This is what happened in v5, so if you want to handle the upload by yourself you can [read the v5 upload docs here](https://backpackforlaravel.com/docs/5.x/crud-fields#upload-1).
 
-The field sends the file, through a Request, to the Controller. The Controller then tries to create/update the Model. That's when the mutator on your model will run. That also means we can do any [file validation](https://laravel.com/docs/5.3/validation#rule-file) (```file```, ```image```, ```mimetypes```, ```mimes```) in the Request, before the file is stored on the disk.
+#### Validation
 
->NOTE: If this field is mandatory (required in validation) please use the [sometimes laravel validation rule](https://laravel.com/docs/10.x/validation#conditionally-adding-rules) together with **required** in your validation. (sometimes|required|file etc... )
+You can use standard Laravel validation rules. But we've also made it easy for you to validate the `upload` fields, using a [Custom Validation Rule](/docs/{{version}}/custom-validation-rules). The `ValidUpload` validation rule allows you to define two sets of rules: 
+- `::field()` - the field rules (independent of the file content);
+- `->file()` - rules that apply to the sent file;
 
-[The ```uploadFileToDisk()``` method](https://github.com/Laravel-Backpack/CRUD/blob/master/src/app/Models/Traits/HasUploadFields.php#L31-L59) will take care of everything for most use cases:
+This helps you avoid most quirks when validating file uploads using Laravel's validation rules.
 
 ```php
-/**
-     * Handle file upload and DB storage for a file:
-     * - on CREATE
-     *     - stores the file at the destination path
-     *     - generates a name
-     *     - stores the full path in the DB;
-     * - on UPDATE
-     *     - if the value is null, deletes the file and sets null in the DB
-     *     - if the value is different, stores the different file and updates DB value
-     * /
-public function uploadFileToDisk($value, $attribute_name, $disk, $destination_path, $fileName = null) {}
-```
-
-If you wish to have a different functionality, you can delete the method call from your mutator and do your own thing.
-
->**The uploaded files are not deleted for you.** When you delete an entry (whether through CRUD or the application), the uploaded files will not be deleted.
+use Backpack\CRUD\app\Library\Validation\Rules\ValidUpload;
 
-If you're NOT using soft deletes on that Model and want the file to be deleted at the same time the entry is, just specify that in your Model's ```deleting``` event:
-```php
-	public static function boot()
-	{
-		parent::boot();
-		static::deleting(function($obj) {
-			\Storage::disk('public_folder')->delete($obj->image);
-		});
-	}
+'image' => ValidUpload::field('required')
+                ->file('file|mimes:jpeg,png,jpg,gif,svg|max:2048'),
```

Comment:
I've noticed the `file` rule isn't _actually_ needed. The way I understand it (without looking at the code) since I specified `->file()`, Backpack will just assume I want to validate it's a `file` too... right? If so, then I don't think we need to document using the `file` validation rule. Quite the contrary, I'd remove it:

```suggestion
                ->file('mimes:jpeg,png,jpg,gif,svg|max:2048'),
```

Reply:
But if we don't document it, I should add this ability to the Validation rule ? Like we do for the `array` rule that gets automatically added, also add the `file` ? `file` is a bit more tricky because it can also be specified with the fluent rule `File` https://laravel.com/docs/10.x/validation#validating-files

Doable, it's just not straightforward like the array rule (if not present, add it). 

Let me know.

Reply:
Hmm let's leave it like this for now... but we might reconsider later on, before launch.